decoder/expr_any_completion_test.go

```diff
@@ -4000,6 +4000,802 @@ func TestCompletionAtPos_exprAny_parentheses(t *testing.T) {
 	}
 }
 
+func TestCompletionAtPos_exprAny_forExpr(t *testing.T) {
+	testCases := []struct {
+		testName           string
+		attrSchema         map[string]*schema.AttributeSchema
+		refTargets         reference.Targets
+		cfg                string
+		pos                hcl.Pos
+		expectedCandidates lang.Candidates
+	}{
+		// list
+		{
+			"list on collection",
+			map[string]*schema.AttributeSchema{
+				"attr": {
+					Constraint: schema.AnyExpression{
+						OfType: cty.List(cty.String),
+					},
+				},
+			},
+			reference.Targets{
+				{
+					Addr: lang.Address{
+						lang.RootStep{Name: "var"},
+						lang.AttrStep{Name: "foo"},
+					},
+					Type: cty.List(cty.String),
+				},
+				{
+					Addr: lang.Address{
+						lang.RootStep{Name: "var"},
+						lang.AttrStep{Name: "mystring"},
+					},
+					Type: cty.String,
+				},
+			},
+			`attr = [for i, v in var: v]
+`,
+			hcl.Pos{Line: 1, Column: 24, Byte: 23},
+			lang.CompleteCandidates([]lang.Candidate{
+				{
+					Label:  "var.foo",
+					Detail: "list of string",
+					Kind:   lang.ReferenceCandidateKind,
+					TextEdit: lang.TextEdit{
+						NewText: "var.foo",
+						Snippet: "var.foo",
+						Range: hcl.Range{
+							Filename: "test.tf",
+							Start:    hcl.Pos{Line: 1, Column: 21, Byte: 20},
+							End:      hcl.Pos{Line: 1, Column: 24, Byte: 23},
+						},
+					},
+				},
+			}),
+		},
+		{
+			"list on value",
+			map[string]*schema.AttributeSchema{
+				"attr": {
+					Constraint: schema.AnyExpression{
+						OfType: cty.List(cty.String),
+					},
+				},
+			},
+			reference.Targets{
+				{
+					Addr: lang.Address{
+						lang.RootStep{Name: "var"},
+						lang.AttrStep{Name: "foo"},
+					},
+					Type: cty.List(cty.String),
+				},
+				{
+					Addr: lang.Address{
+						lang.RootStep{Name: "var"},
+						lang.AttrStep{Name: "bar"},
+					},
+					Type: cty.String,
+				},
+			},
+			`attr = [for i, v in var: v]
```

Comment:
I found this confusing at first, because I was expecting mainly `v` to be used here. But after giving it some thought, it makes sense to have the full completion list here.

Reply:
The `v` as in the value reference support is something now tracked under https://github.com/hashicorp/terraform-ls/issues/1608